src/Arcus.Security.Providers.AzureKeyVault/Configuration/SecretStoreBuilderExtensions.cs

```diff
@@ -42,18 +73,113 @@ public static class SecretStoreBuilderExtensions
         /// <param name="rawVaultUri">The Uri of the Azure Key Vault you want to connect to.</param>
         /// <param name="connectionString">The connection string to use to authenticate, if applicable.</param>
         /// <param name="azureADInstance">The azure AD instance to use to authenticate, if applicable.</param>
+        /// <param name="allowCaching">The flag to indicate whether to include caching during secret retrieval in Azure key vault.</param>
         public static SecretStoreBuilder AddAzureKeyVaultWithManagedServiceIdentity(
             this SecretStoreBuilder builder,
             string rawVaultUri,
             string connectionString = null,
-            string azureADInstance = null)
+            string azureADInstance = null,
+            bool allowCaching = false)
         {
             Guard.NotNull(rawVaultUri, nameof(rawVaultUri));
 
             return AddAzureKeyVault(
                 builder,
                 new ManagedServiceIdentityAuthentication(connectionString, azureADInstance),
-                new KeyVaultConfiguration(rawVaultUri));
+                new KeyVaultConfiguration(rawVaultUri),
+                allowCaching);
+        }
+
+        /// <summary>
+        /// Adds Azure Key Vault as a secret source which uses Managed Identity authentication.
+        /// </summary>
+        /// <param name="builder">The builder to create the secret store.</param>
+        /// <param name="rawVaultUri">The Uri of the Azure Key Vault you want to connect to.</param>
+        /// <param name="cacheConfiguration">The configuration to control how the caching will be done.</param>
+        public static SecretStoreBuilder AddAzureKeyVaultWithManagedServiceIdentity(
```

Comment:
I'm thinking out loud but shouldn't we align with the overload above and use optional except for the cache configuration?

Ok with splitting into dedicated methods, but then we should maybe do something similar for the one above?

Reply:
Move back to one and moved the cache config before the optional params.